airflow/kubernetes/pod_generator.py

```diff
@@ -459,10 +459,14 @@ def make_unique_pod_id(pod_id: str) -> str:
             return None
 
         safe_uuid = uuid.uuid4().hex  # safe uuid will always be less than 63 chars
-        # Strip trailing '-' and '.' as they can't be followed by '.'
-        trimmed_pod_id = pod_id[:MAX_LABEL_LEN].rstrip('-.')
 
-        safe_pod_id = f"{trimmed_pod_id}.{safe_uuid}"
+        # Get prefix length after subtracting the uuid length. Clean up . and - from
+        # end of podID because they can't be followed by '.'
```

Comment:
```suggestion
        # Get prefix length after subtracting the uuid length. Clean up '.' and '-' from
        # end of podID ('.' can't be followed by '-').
```

Technically we no longer need to remove any trailing`-`, but I think it makes sense to continue doing so.